### PR TITLE
Check if the cells contain valid values before validation

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed** 
This PR finetunes on how cells are validated in the spreadsheet. 
Currently we only checks if a mandatory cell has a value, not what kind of value it is if a selection of values is possible. 

This PR fixes addressing and checking the following cases for the value within a mandatory cell: 

1.) If only one value is expected, ensure that the expected value was provided 
2.) If one of many value is expected, ensure that the provided value is one of the expected values
3.) If no specific value is expected, ensure that the cell was not left blank

If the cell is not mandatory, all kind of values are accepted.

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java [hunk 1]
```diff
@@ -378,9 +378,8 @@ private void generateGenomicsSheet() {
    * @return cellValueOptionsForColumnMap map grouping the selectable cell values within the cells
    * of a column with the {@link SamplesheetHeaderName} of the colum
    */
-  private LinkedHashMap<SamplesheetHeaderName, List<String>> cellValueOptionsForCellWithinColumn(
+  private void generateCellValueOptionsMap(
```
Comment (wow-such-code) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:381]: Is this correctly reset when an experiment is changed?

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java [hunk 2]
```diff
@@ -474,10 +472,12 @@ public Result<Void, InvalidSpreadsheetInput> areInputsValid() {
       // mandatory not filled in --> invalid
       for (int colId : mandatoryInputCols) {
         Cell cell = row.getCell(colId);
-        if (SpreadsheetMethods.cellToStringOrNull(cell).isBlank()) {
-          invalidCells.add(cell);
-        } else {
+
+        if (isCellValueValid(cell)) {
           validCells.add(cell);
```
Comment (wow-such-code) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:477]: We have to be careful with the two PRs here, the other PR changes the following:

          // if a background color was set, but the cell is valid, we need to change the style
          if(cell.getCellStyle().getFillBackgroundColorColor()!=null) {
            validCells.add(cell);
          }

Comment (wow-such-code) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:477]: just fyi

**Review by wow-such-code — CHANGES_REQUESTED** [review]
looking good, but I have a question :)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java (hunk 2 above)
Comment (wow-such-code) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java]: I think I resolved it

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java (hunk 1 above)
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:384]: Yes! 🙌

**Review by wow-such-code — APPROVED** [review]
do it

**Pull request merged**